Fix flaky `TestTruncation.truncation_discrete_random`

mattiadg opened the pull request:
Closes #6206
* Add seed to rejection test
* Add a comment about the test probability

<!-- !! Thank your for opening a PR !! -->

**What is this PR about?**
Fix Potential flaky test: truncation_discrete_random
I add a seed to the test part that can fail stochastically.

**Checklist**
+ [x] Explain important implementation details 👆
+ [x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [x] Are the changes covered by tests and docstrings?
+ [x] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...

## Bugfixes / New features
- ...

## Docs / Maintenance
- Fix flaky `TestTruncation.truncation_discrete_random`

mattiadg committed:
Fix Potential flaky test: truncation_discrete_random
#6206
* Add seed to rejection test
* Add a comment about the test probability

ricardoV94 commented on pymc/tests/distributions/test_truncated.py:
```diff
@@ -181,7 +181,11 @@ def test_truncation_discrete_random(op_type, lower, upper):
+            # = (1 - 0.2 ** 3) ** 500 = 0.018
+            # Still, this probability can be too high to make this test pass with any seed.
+            random_seed = 2297228
+            draw(xt, random_seed=np.random.RandomState(random_seed))
```
Assuming it passes with that seed, you can simply do:
```suggestion
            draw(xt, random_seed=2297228)
```

mattiadg replied:
`draw` requires as input a RandomState, not an int 
https://github.com/pymc-devs/pymc/blob/main/pymc/sampling.py#L2269-L2274

ricardoV94 replied:
Draw accepts int, RandomState or Generator. That type hint is a union of these 3

https://github.com/pymc-devs/pymc/blob/b5b63f5d6c8d5891382cce1af9bd5ea22de795cf/pymc/sampling.py#L107

If anything we should use Generator because RandomState is legacy numpy at this point. But integer is fine for this test.

mattiadg committed:
Update pymc/tests/distributions/test_truncated.py

Pass a random seed to draw() for deterministic result instead of a np.random.RandomState

Co-authored-by: Ricardo Vieira <28983449+ricardoV94@users.noreply.github.com>
